### PR TITLE
Dnsname build fixes

### PR DESCRIPTION
**Pull request by pieterlexis**
This fixes the build issues we had with the tools. Also, it disables the broken lmdb backend in the debian buiulds (so we can actually build packages).

### CODE REVIEW

**Comment by pieterlexis** [comment]
Needs review.

File: .travis.yml [hunk 1]
```diff
@@ -90,6 +90,7 @@ script:
      --enable-remotebackend-zeromq
      --enable-experimental-ed25519
      --enable-experimental-pkcs11
+     --enable-tools
```
Comment (mind04) [inline .travis.yml]: all tools are build on line 106 and if you enable the toolsbuild there you must remove line 108 (all marked with #DNSName)

File: pdns/dnsreplay.cc [hunk 2]
```diff
@@ -280,7 +280,7 @@ bool isRootReferral(const MOADNSParser::answers_t& answers)
     //    cerr<<(int)iter->first.d_place<<", "<<iter->first.d_label<<" "<<iter->first.d_type<<", # "<<answers.size()<<endl;
     if(iter->first.d_place!=2)
       ok=false;
-    if(iter->first.d_label!="." || iter->first.d_type!=QType::NS)
+    if(iter->first.d_label!=DNSName(".") || iter->first.d_type!=QType::NS)
```
Comment (mind04) [inline pdns/dnsreplay.cc]: if (!iter->first.d_label.isRoot() || ....

**Comment by pieterlexis** [comment]
Now that travis is happy, needs review.

File: pdns/Makefile.am [hunk 3]
```diff
@@ -952,6 +957,8 @@ dnsdemog_LDADD = $(POLARSSL_LIBS)
 
 rec_control_SOURCES = \
 	arguments.cc arguments.hh \
+	dnsname.hh dnsname.cc \
+	dnslabeltext.cc \
```
Comment (Habbie) [inline pdns/Makefile.am]: Is this still necessary now that Logger<<DNSName was moved to dnsname.hh/cc?

File: pdns/dnsscope.cc [hunk 4]
```diff
@@ -431,7 +431,7 @@ try
 	      rem.sin4.sin_port=0;
 
 	      if(doServFailTree)
-		root.submit(mdp.d_qname, mdp.d_header.rcode, rem);
+		root.submit(mdp.d_qname.toString(), mdp.d_header.rcode, rem);
```
Comment (Habbie) [inline pdns/dnsscope.cc]: Suggestion (could be for a later round): make StatNode take DNSName

File: pdns/speedtest.cc [hunk 5]
```diff
@@ -505,18 +484,9 @@ static bool magicAddrMatch(const QType& query, const QType& answer)
...
-  int counta=(a!=dot), countb=(b!=dot);
-  
-  for(string::size_type n=0;n<a.size();++n)
-    if(a[n]=='.')
-      counta++;
-  for(string::size_type n=0;n<b.size();++n)
-    if(b[n]=='.')
-      countb++;
-  return counta>countb;
+  return (a.isPartOf(b) && a.countLabels() > b.countLabels());
```
Comment (Habbie) [inline pdns/speedtest.cc]: The old code did not do the equivalent of `isPartOf`, is it okay that you changed this?

**Comment by Habbie** [comment]
Nits posted. LGTM otherwise.

File: pdns/Makefile.am (hunk 3 above)
Comment (pieterlexis) [inline pdns/Makefile.am]: yes, several included files use `DNSName` functions.

**Comment by pieterlexis** [comment]
Habbie's nits are fixed. @mind04?

**Comment by mind04** [comment]
LGTM

**Pull request merged**